python/stempy/image/__init__.py

```diff
@@ -1,9 +1,40 @@
 from stempy import _image
+from stempy import _io
 import numpy as np
-
-def create_stem_images(reader, inner_radii,
+import h5py
+
+
+def get_hdf5_reader(h5file):
+    # the initialization is at the io.cpp
+    dset_frame=h5file['frames']
+    dset_frame_shape=dset_frame.shape
+    totalImgNum=dset_frame_shape[0]
+    imagewidth=dset_frame_shape[1]
+    imageheight=dset_frame_shape[2]
+    
+    dset_stem_shape=h5file['stem/images'].shape
+    scanwidth=dset_stem_shape[1]
+    scanheight=dset_stem_shape[2]
+    
+    # the number here are same with the smallscan data
+    blocksize=32
+    blockNumInFile=25
+    image_numbers = []
+    
+    h5reader = _io._h5reader(dset_frame, image_numbers, imagewidth, imageheight, scanwidth, scanheight, blocksize, blockNumInFile, totalImgNum)
+    return h5reader
+
+def create_stem_images(input, inner_radii,
                        outer_radii, width=0, height=0,
                        center_x=-1, center_y=-1):
+    # check if the input is the hdf5 dataset
+    if(isinstance(input, h5py._hl.files.File)):
```

Comment:
I think we can broaden this check to `hasattr(input, '__len__') and hasattr(input, '__getitem__')`, then is should also work with regular numpy arrays. We should then remain some the of the classes so they don't explicitly refer to HD5, for example `H5Reader` would become `SequenceReader`.